ports/servoshell/desktop/webxr.rs

```diff
@@ -60,14 +62,29 @@ impl XrDiscoveryWebXrRegistry {
     }
 }
 
+struct XrPrefObserver(Arc<AtomicBool>);
+
+impl prefs::Observer for XrPrefObserver {
+    fn prefs_changed(&self, changes: Vec<(&'static str, prefs::PrefValue)>) {
+        if let Some((_, value)) = changes.iter().find(|(name, _)| *name == "dom_webxr_test") {
+            let prefs::PrefValue::Bool(value) = value else {
+                return;
+            };
+            self.0.store(*value, Ordering::Relaxed);
+        }
+    }
+}
+
 #[cfg(feature = "webxr")]
 impl WebXrRegistry for XrDiscoveryWebXrRegistry {
     fn register(&self, xr: &mut servo::webxr::MainThreadRegistry) {
         use servo::webxr::headless::HeadlessMockDiscovery;
 
-        if pref!(dom_webxr_test) {
-            xr.register_mock(HeadlessMockDiscovery::default());
```

Comment:
I spent a while looking at dynamically registering and unregistering the mock, but the MainThreadRegistry is only stored in the compositor after startup, and adding entirely new embedding APIs just for this did not feel justified.